File: java/sprykerFileCreator/model/definition/spryker/YvesDependencyProviderDefinition.java
package sprykerFileCreator.model.definition.spryker;

import pav.sprykerFileCreator.model.definition.ClassDefinitionInterface;
import pav.sprykerFileCreator.model.generator.SprykerConstants;

public class YvesDependencyProviderDefinition implements ClassDefinitionInterface {

    @Override
    public String getClassType() {
        return SprykerConstants.YVES_DEPENDENCY_PROVIDER;
    }

    @Override
    public String getNamePattern() {
        return "{{ bundleName }}DependencyProvider";
    }

    @Override
    public String getNamespacePattern() {
        return "{{ projectName }}\\Yves\\{{ bundleName }}";
    }

    @Override
    public String getMethodForReturnType() {
        return "";
    }

    @Override
    public String[] getDocBlockClasses() {
        String [] docBlockClasses = {};

        return docBlockClasses;
    }

    @Override
    public String getDefaultParentClass() {
        return "Spryker\\Yves\\Kernel\\AbstractBundleDependencyProvider";
    }
}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/definitions/spryker/YvesDependencyProviderDefinition.kt
package com.github.tobi812.sprykerplugin.models.definitions.spryker

import com.github.tobi812.sprykerplugin.constants.SprykerConstants
import com.github.tobi812.sprykerplugin.models.definitions.ClassDefinitionInterface

class YvesDependencyProviderDefinition : ClassDefinitionInterface {
    override val classType: String
        get() = SprykerConstants.YVES_DEPENDENCY_PROVIDER
    override val namePattern: String
        get() = "{{ bundleName }}DependencyProvider"
    override val namespacePattern: String
        get() = "{{ projectName }}\\Yves\\{{ bundleName }}"
    override val methodForReturnType: String
        get() = ""
    override val docBlockClasses: Array<String>
        get() = arrayOf()
    override val defaultParentClass: String
        get() = "Spryker\\Yves\\Kernel\\AbstractBundleDependencyProvider"
}